core/src/main/scala/org/bitcoins/core/wallet/utxo/InternalAddressTag.scala

```diff
@@ -0,0 +1,144 @@
+package org.bitcoins.core.wallet.utxo
+
+/**
+  * An AddressTagNames that is native to Bitcoin-S.
+  * InternalAddressTagNames are still usable when using Bitcoin-S
+  * as a dependency
+  */
+sealed trait InternalAddressTagName extends AddressTagName
+
+/**
+  * An AddressTagType that is native to Bitcoin-S.
+  * InternalAddressTagTypes are still usable when using Bitcoin-S
+  * as a dependency
+  */
+sealed trait InternalAddressTagType extends AddressTagType
+
+/**
+  * An AddressTag that is native to Bitcoin-S.
+  * InternalAddressTags are still usable when using Bitcoin-S
+  * as a dependency
+  */
+sealed trait InternalAddressTag extends AddressTag
+
+/** An unknown address tag name, should mostly be used for parsing external address tags */
+case class UnknownAddressTagName(name: String) extends InternalAddressTagName {
+  require(InternalAddressTagName.fromStringOpt(name).isEmpty,
+          s"This tag name is already defined, got $name")
+}
+
+/** An unknown address tag type, should mostly be used for parsing external address tags */
+case class UnknownAddressTagType(typeName: String)
+    extends InternalAddressTagType {
+  require(InternalAddressTagType.fromStringOpt(typeName).isEmpty,
+          s"This tag type is already defined, got $typeName")
+}
+
+/** An address tag without an unknown type, should mostly be used for parsing external address tags */
+case class UnknownAddressTag(tagName: AddressTagName, tagType: AddressTagType)
```

Comment:
Why do we have an `UnknownAddressTag extend InternalAddressTag`? If it's an internal tag, we should be able to define it. Only external things should have the option for `UnkonwnAddressTag` no?

Reply:
If that is the case, I have two suggestions: 

1. Rename it to `ExternalAddressTagWrapper` or something like that
2. Redefine it to take an `ExternalAddressTag` as a parameter i.e. 

```scala
case class ExternalAddressTagWrapper(external: ExternalAddressTag)
    extends InternalAddressTag {
  override val tagName: AddressTagName = external.tagName
  override val tagType: AddressTagType = external.tagType
}
```